code/modules/food&drinks/drinks/drinks.dm

```diff
@@ -288,3 +298,302 @@
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/orangeaid
+	name = "Orange-Aid"
+	desc = "A vitamin tonic that promotes good eyesight and health."
+	icon_state = "starkist"
+	New()
+		..()
+		reagents.add_reagent("orangejuice", 20)
+		reagents.add_reagent("oculine", 20)
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/limeaid
+	name = "Lime-Aid"
+	desc = "Antihol mixed with lime juice. A well-known cure for hangovers."
+	icon_state = "space-up"
+	New()
+		..()
+		reagents.add_reagent("limejuice", 20)
+		reagents.add_reagent("antihol", 20)
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/runoff
+	name = "Spooky Dan's Runoffa Cola"
+	desc = "A spoooky cola for Halloween!  Rumors that Runoff Cola contains actual industrial runoff are unsubstantiated."
+	icon_state = "purple_can"
+	New()
+		..()
+		reagents.add_reagent("chlorine", 5)
+		reagents.add_reagent("phosphorus", 5)
+		reagents.add_reagent("mercury", 5)
+		reagents.add_reagent("vhfcs", 10)
+		reagents.add_reagent("cola", 15)
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/horror
+	name = "Spooky Dan's Horrortastic Cola"
+	desc = "A terrifying Halloween soda.  It's especially frightening if you're diabetic."
+	icon_state = "glowingbeer"
+	New()
+		..()
+		reagents.add_reagent("ectoplasm", 10)
+		reagents.add_reagent("sulfur", 5)
+		reagents.add_reagent("vhfcs", 5)
+		reagents.add_reagent("cola", 20)
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/grones
+	name = "Grones Soda"
+	desc = "Bursting with 10 different flavors!"
+	icon_state = "grones"
+	New()
+		..()
+		reagents.add_reagent("cola", 20)
+		reagents.add_reagent(pick("atropine","serotrotium","lithium","mercury","charcoal","diphenhydramine","spaceacillin","hairgrownium","ephedrine","capsaicin"), 10)
+
+/obj/item/weapon/reagent_containers/food/drinks/soda_cans/decirprevo
+	name = "Decirprevo Bottled Water"
+	desc = "Bottled from our cool natural springs on Europa."
+	icon_state = "waterbottle"
+	New()
+		..()
+		reagents.add_reagent("water", 45)
+		reagents.add_reagent("iodine", 5)
+
+////Stinkeye///   Holy buns don't ever drink this -Fox
+
+/obj/item/weapon/reagent_containers/food/drinks/stinkeye
+	name = "Stinkeye's Special Reserve"
+	desc = "An old bottle labelled 'The Good Stuff'. This probably has enough kick to knock an elephant on its ass."
+	icon_state = "whiskeybottle"
+	volume = 250
+	New()
+		..()
+		reagents.add_reagent("beer", 30)
+		reagents.add_reagent("wine", 30)
+		reagents.add_reagent("cider", 30)
+		reagents.add_reagent("vodka", 30)
+		reagents.add_reagent("ethanol", 30)
+		reagents.add_reagent("eyenewt", 30)
+
+////Discount Dan's Soup//////       May God have mercy on your souls---and stomachs.    -Fox
+
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup
+	name = "Discount Dan's Quik-Noodles"
+	desc = "A self-heating cup of noodles. There's enough sodium in these to put the Dead Sea to shame."
+	icon_state = "dansoup"
+	volume = 65
+	var/selfheat = 0
```

Comment:
These already exist, have you even looked in a food dispenser lately?

(although they work differently)